pkg/schedule/schedulers/hot_region.go

```diff
@@ -1347,6 +1356,22 @@ func (bs *balanceSolver) buildOperators() (ops []*operator.Operator) {
 	targetLabel := strconv.FormatUint(dstStoreID, 10)
 	dim := bs.rankToDimString()
 
+	splitRegions := make([]*core.RegionInfo, 0)
+	if bs.opTy == movePeer {
+		for _, region := range []*core.RegionInfo{bs.cur.region, bs.cur.revertRegion} {
+			if region == nil {
+				continue
+			}
+			if region.GetApproximateSize() > bs.GetOpts().GetMaxMovableHotPeerSize() {
+				hotSchedulerNeedSplitBeforeScheduleCounter.Inc()
+				splitRegions = append(splitRegions, region)
+			}
+		}
+	}
+	if len(splitRegions) > 0 {
+		return bs.createSplitOperator(splitRegions)
+	}
```

Comment:
Should we move it to the beginning of this function?

Reply:
Good suggestion.